Fixes to help make VisualFSharp.sln more usable

dsyme opened the pull request:
The most important thing is `  <EnableDefaultEmbeddedResourceItems>false</EnableDefaultEmbeddedResourceItems>` in VisualFSharp.UnitTests.fsproj. The other changes use more normal ways to specify the resources, and ensure we restore test projects with `build vs`

With this was able to 

* msbuild VisualFSharp.sln
* build VisualFSharp.sln from VS
* F5 launch into VisualFSharpFull.vsix.

However this doesn't solve the endless rebuild of FSHarp.Editor, which really kills the usability of VisualFSharp.sln when hitting F5

@brettfo please take a look.

dsyme commented on vsintegration/src/FSharp.Editor/FSharp.Editor.fsproj:
```diff
@@ -20,7 +21,7 @@
 
   <ItemGroup>
-    <EmbeddedResource Update="FSharp.Editor.resx">
+    <EmbeddedResource Include="FSharp.Editor.resx">
```
@brettfo I'm pretty sure this is normal, I think you were only using Update to avoid EnableDefaultEmbeddedResourceItems?

brettfo replied:
Correct, I was using `Update` because `$(EnableDefault...)` was true.  What's the benefit to specifying it this way, e.g., to not allowing the `**/*.resx` glob from running?  The default behavior for SDK projects is to auto-include all `*.resx` files into the `@(EmbeddedResource)` item collection.  I was trying to stay in line with that so that if another `*.resx` file is added to this directory that it's automatically included in the group so we don't have to manually manage the files in the project.

brettfo commented on vsintegration/fsharp-vsintegration-unittests-build.proj:
```diff
@@ -31,6 +31,7 @@
   <Import Project="..\src\root.traversal.targets"/>
 
   <Target Name="Restore">
+    <MSBuild Projects="@(ProjectFiles)" Targets="Restore" />
```
If you're going to restore everything then remove all of the `<SdkProjects>` lines; those only existed to restore a subset of the projects.

dsyme replied:
agreed we don't need this

brettfo commented on vsintegration/src/FSharp.LanguageService.Base/FSharp.LanguageService.Base.csproj:
```diff
@@ -23,7 +24,7 @@
 
   <ItemGroup>
-    <EmbeddedResource Update="Microsoft.VisualStudio.Package.LanguageService.resx">
+    <EmbeddedResource Include="Microsoft.VisualStudio.Package.LanguageService.resx">
```
Same `Update` vs. `Include` comment as above.

dsyme replied:
As mentioned below, the default globbing was incorrectly picking up resources from a project in `MockTypeProviders` subfolder after you built those projects

brettfo commented on vsintegration/src/FSharp.LanguageService/FSharp.LanguageService.fsproj:
```diff
@@ -28,7 +29,7 @@
   <ItemGroup>
     <!-- VSPackage.resx already included -->
-    <EmbeddedResource Update="FSLangSvcStrings.resx">
+    <EmbeddedResource Include="FSLangSvcStrings.resx">
```
Ditto.

brettfo commented on vsintegration/tests/Salsa/VisualFSharp.Salsa.fsproj:
```diff
@@ -8,6 +8,7 @@
     <NoWarn>$(NoWarn);45;47;52;58;75</NoWarn>
     <DisableImplicitFSharpCoreReference>true</DisableImplicitFSharpCoreReference>
     <UsePackageTargetFallbackHack>true</UsePackageTargetFallbackHack>
+    <EnableDefaultEmbeddedResourceItems>false</EnableDefaultEmbeddedResourceItems>
```
What's the benefit, particularly with this project and `VisualFSharp.UnitTests` where there are no `*.resx` files?  If the `**/*.resx` glob is breaking incremental build then that's a larger issue we should address at the SDK level.

dsyme replied:
@brettfo VisualFSharp.UnitTests has projects in sub-directories which put resources in `obj/Debug` directories when built, and the implicit glob was picking those up

brettfo replied:
I didn't know about the `obj\Debug` issue.  I see that you rearranged the directories which I like.  Only one final nit-pick; instead of decorating each project with `<EnableDefault...>false</>` can you add it to an appropriate upstream `Directory.Build.props` or `FSharp.Directory.Build.props`?  That way we won't forget to add it to any new projects in those subdirectories.

KevinRansom commented:
@brettfo I had the same question, but came down on the side of meh ... at least this way it is explicit.

dsyme commented:
I think I don't mind either way. I couldn't find any MSBuild documentation for `Update=` and had no goodd mental model for how it worked so assumed we were using some internal thing.  

And I needed to use `EnableDefaultEmbeddedResourceItems` in one project because the implicit resource glob was incorrect for "VisualFSharp.UnitTests" (incorrectly getting a completely unrelated resource from MockTypeProviders), so I basically no longer trusted implicit.

Are we actually relying on the implicit glob anywhere?

dsyme commented:
@brettfo @KevinRansom I've also 

* moved vsintegration/tests/unittests/MockTypeProviders --> vsintegration/tests/MockTypeProviders so we remove the nested projects 
* renamed vsintegration/tests/unittests --> vsintegration/tests/VisualFSharp.UnitTests

Those renames/moves are enough to solve the problem

![image](https://user-images.githubusercontent.com/7204669/37521555-114ab6a8-2919-11e8-95c3-c8e8b90d2342.png)

dsyme commented:
@brettfo I'll merge this if that's ok - I've mentioned above why moving to implicit globbing was a problem for us

dsyme commented:
@brettfo Sorry, I didn't see this comment for some reason:

> Only one final nit-pick; instead of decorating each project with <EnableDefault...>false</> can you add it to an appropriate upstream Directory.Build.props or FSharp.Directory.Build.props? That way we won't forget to add it to any new projects in those subdirectories.

Agreed we should do that
